src/lex.cpp

```diff
@@ -722,16 +734,37 @@ const char8* lexer::end_of_previous_token() const noexcept {
 }
 
 void lexer::parse_hexadecimal_number() {
-  QLJS_ASSERT(this->is_hex_digit(this->input_[0]) || this->input_[0] == '.');
   char8* input = this->input_;
 
   while (this->is_hex_digit(*input)) {
     input += 1;
   }
-
   this->input_ = check_garbage_in_number_literal(input);
 }
 
+char8* lexer::check_garbage_in_number_literal(char8* input) {
```

Comment:
Minor: Un-move this function so the Git-diff is easier to read.

Reply:
Moved to seperate commit

Reply:
hey